### DISCUSSION TITLE
Add background in a transparent PNG image

### DISCUSSION THREAD

**Discussion by ppc2017** [lllyasviel/Fooocus#2463] [Q&A]
I have a transparent PNG image and I want to add a background. I tried inpaint but I have to draw the background area with the brush.
Since the PNG image contains the alpha channel, is there any way to use it without the need to use the brush?

I know the mask upload option, but I have to generate the alpha mask in Photoshop and I want to avoid this step.

**Comment by mashb1t** [comment]
There is, use automated mask generation in PR https://github.com/lllyasviel/Fooocus/pull/2032 or directly use my fork https://github.com/mashb1t/Fooocus.
=> basically no need to generate masks ever again, as you can also prompt your mask.
Switching guide: https://github.com/mashb1t/Fooocus?tab=readme-ov-file#switching-from-fooocus-to-this-fork

The PR is most likely going to be merged in 2.3.0, ETA still TBD.
